src/analytics.js

```diff
@@ -56,3 +57,28 @@ export function triggerAnalyticsEvent(nodeOrDoc, eventType, opt_vars) {
     analytics.triggerEvent(eventType, opt_vars);
   });
 }
+
+/**
+ * Helper method to create analytics element for specific root.
+ * @param {BaseElement} baseInstance
+ * @param {!JSONType} config
+ */
+export function createAnalyticsForExtension(baseInstance, config) {
+  // Note: Require ampdoc include analytics script
+  analyticsForDocOrNull(baseInstance.element).then(analytics => {
+    if (!analytics) {
+      return;
+    }
+    // Create analytics element;
+    const analyticsElem =
+        baseInstance.win.document.createElement('amp-analytics');
+    const scriptElem = createElementWithAttributes(baseInstance.win.document,
+        'script', {
+          'type': 'application/json',
+        });
+    scriptElem.textContent = JSON.stringify(config);
```

Comment:
What I planned to do is with `scoped` is: always create new analytics root for scoped <amp-analytics> element.

The JSON thing. Sorry I don't quite understand. I think what you mean is we can avoid the `stringify` then `parse` in this case. However since I am re using the old code here https://github.com/ampproject/amphtml/blob/master/extensions/amp-analytics/0.1/amp-analytics.js#L357, I think I will have to stringify it at this point. But I agree we should avoid this.